src/emit.h

```diff
@@ -0,0 +1,71 @@
+#ifndef EMIT_H
+#define EMIT_H
+
+#include "json.h"
+#include "string_id.h"
+#include "field.h"
+
+class emit;
+using emit_id = string_id<emit>;
+
+class emit
+{
+    public:
+        emit() : id_( NULL_ID ) {}
+
+        const emit_id &id() const {
+            return id_;
+        }
+
+        /** When null @ref field is always fd_null */
+        bool is_null() const {
```

Comment:
If you want an interface, please don't reimplement its guts (even if they're tiny): 
```c++
bool is_null() const {
    return id.is_null();
}
```